### PR TITLE
[CDAP-11334] Adds column level directives 

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Modifies the layout for right panel to have columns opened by default and have the column information in a more tabular way
- Adds Column Actions dropdown to apply column level directives
  - Bulkset (`set columns <column-names>`)
  - Cleanse columns

JIRA:
https://issues.cask.co/browse/CDAP-11333
https://issues.cask.co/browse/CDAP-11334
https://issues.cask.co/browse/CDAP-11437

Build: https://builds.cask.co/browse/CDAP-DRC6480-2

### CODE REVIEW

**Comment by ajainarayanan** [comment]
Talked @elfenheart about getting summary. Will rebase on top of @elfenheart's changes once its merged.

File: cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/ColumnsTab.scss [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  &.empty-search-container {
+    height: auto;
+  }
+
+  .empty-search {
+    font-size: 18px;
+    font-weight: 500;
+    margin-left: auto;
+    margin-right: auto;
+    margin-top: 25vh;
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/ColumnsTab.scss:133]: using `vh` might be a little dangerous here with the height of container being different in pipeline mode?

File: cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/index.js [hunk 2]
```diff
@@ -45,58 +40,42 @@ export default class ColumnsTab extends Component {
...
-        this.setState({
-          selectedHeaders: newState.selectedHeaders
-        });
-      }
-      this.getSummary();
+      let columnInfo = DataPrepStore.getState().columnsInformation;
+      this.setState({
+        selectedHeaders: newState.selectedHeaders,
+        columns: columnInfo.columns,
+        headers: newState.headers.map((res) => {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/index.js]: we should initialize the state as well at line 35 with these values.

File: cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/index.js [hunk 3]
```diff
@@ -229,7 +187,14 @@ export default class ColumnsTab extends Component {
       );
     }
 
-    let displayHeaders = this.state.headers;
+    let index = -1;
+    let displayHeaders = this.state.headers.map(header => {
```
Comment (elfenheart) [inline cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/index.js:193]: isn't index being passed as second argument to the callback of `.map()`?

File: cdap-ui/app/cdap/text/text-en.yaml [hunk 4]
```diff
@@ -333,11 +333,42 @@ features:
...
     Directives:
       apply: Apply
       cancel: Cancel
+      ColumnActions:
+        label: Column Actions
+        Bulkset:
+          modalTitle: Bulk set column names
+          setBtnLabel: Set Columns
+          description: Enter column names in order starting from the first column
+          textareaplaceholder: "Enter column names seperated by comma. No special character (eg., @ - #) are allowed"
```
Comment (elfenheart) [inline cdap-ui/app/cdap/text/text-en.yaml]: alphabetically sort these changes

File: cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/index.js (hunk 3 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/index.js:193]: Its not the same index. This index calculation skips the `isDetail` rows that gets added to headers list when user clicks on any row with row info.

File: cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/ColumnsTab.scss (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/DataPrep/DataPrepSidePanel/ColumnsTab/ColumnsTab.scss:133]: The margin top for columns tab(dataprep container) in both the cases are the same. Shouldn't be different in either cases. Let me know if you notice anything different between dataprep and pipelines studio or if the styling can be simpler.

**Comment by elfenheart** [comment]
screen shot 2017-05-15 at 3 09 16 pm

the caret on the column actions seemed to be cutoff.

Also on the dropdown items for the column actions, perhaps add a pointer cursor?

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by elfenheart** [comment]
👍  LGTM

**Review by bdmogal — APPROVED** [review]

**Pull request merged**